src/__tests__/automations.test.ts

```diff
@@ -0,0 +1,80 @@
+import axios from "axios";
+import MockAdapter from "axios-mock-adapter";
+import { CourierClient } from "../index";
+
+import { ICourierAutomationInvokeResponse } from "../automations/types";
+
+const mockAutomationInvokeResponse: ICourierAutomationInvokeResponse = {
+  runId: "1234"
+};
+
+describe("CourierAutomations", () => {
+  let mock: MockAdapter;
+
+  beforeEach(() => {
+    mock = new MockAdapter(axios);
+    mock.onPost("/automations/invoke").reply(200, mockAutomationInvokeResponse);
+    mock
+      .onPost(/\/automations\/.*\/invoke/)
+      .reply(200, mockAutomationInvokeResponse);
+  });
+
+  test(".invokeAdHocAutomation", async () => {
+    const { automations } = CourierClient({
+      authorizationToken: "AUTH_TOKEN"
+    });
+
+    await expect(
+      automations.invokeAdHocAutomation({
+        automation: {
+          cancelation_token: "I_AM_TOKEN",
+          steps: [
+            {
+              action: "send"
+            },
+            {
+              action: "cancel"
+            },
+            {
+              action: "send-list",
+              list: "my-list"
+            },
+            {
+              action: "delay"
+            }
+          ]
+        },
+        brand: "BRAND_ID",
+        data: {
+          example: "EXAMPLE_DATA"
+        },
+        profile: {
+          email: "foo@bar.com"
+        },
+        recipient: "RECIPIENT_ID",
+        template: "TEMPLATE_NAME_OR_ID"
+      })
+    ).resolves.toMatchObject(mockAutomationInvokeResponse);
+  });
+
+  test(".invokeAutomationTemplate", async () => {
+    const { automations } = CourierClient({
+      authorizationToken: "AUTH_TOKEN"
+    });
+
+    await expect(
```

Comment:
Are there any error situation that are worth putting in unit tests?

Reply:
Not really. We throw a 400 if required attributes aren't passed along which TS would take care of, and 5XX would probably be in axios error handling spectrum.